backend/src/packages/shifts/shift.controller.ts

```diff
@@ -39,7 +39,7 @@ class ShiftController extends Controller {
     this.addRoute({
       path: ShiftsApiPath.ROOT,
       method: 'POST',
-      authStrategy: AuthStrategy.INJECT_USER,
+      authStrategy: [AuthStrategy.VERIFY_JWT, AuthStrategy.INJECT_USER],
       validation: {
```

Comment:
It seems that difference between AuthStrategy.VERIFY_JWT and AuthStrategy.INJECT_USER that first throws error on unauthorized user and the second one, so we need to use only one of them

Reply:
It turns out that INJECT_USER only attach user to request. And if there isn't token, it attaches user=undefined and allowed to get to the private endpoints.

Reply:
Then we can use only AuthStrategy.VERIFY_JWT

Reply:
Yes, you are right. Thank you.